lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2005,10 +2005,6 @@ bool llvm::writeSpirv(Module *M, const SPIRV::TranslatorOpts &Opts,
 }
 
 bool llvm::regularizeLlvmForSpirv(Module *M, std::string &ErrMsg) {
-  std::unique_ptr<SPIRVModule> BM(SPIRVModule::createSPIRVModule());
```

Comment:
@svenvh, could you please pay attention to this particular change?

When I started working on this PR these lines looked like:
```c++
  std::unique_ptr<SPIRVModule> BM(SPIRVModule::createSPIRVModule());
```

And `BM` was unused, so - I deleted it. Right now it looks like `BM` is used to emit error messages and probably it is better to leave this code as-is

Reply:
Yes it is better to keep `BM` as we use the error log to report any issues found by `isValidLLVMModule`; you are (unintentionally?) deleting that check actually.

Reply:
Thanks, I will leave this code as-is

Reply:
Restored this code and added test for it